# Github repositories
Participants in the group worked on the weekly exercises alone prior to forming groups:

Mikel's repository: https://github.com/Mikel-Yu/comsocsci2026_private

Poul's repository: https://github.com/poulguoskov/cososc

# Assignment 1

## Part 1: Ready Made vs Custom Made Data

### 1. What are pros and cons of the custom-made data used in Centola's experiment and the ready-made data used in Nicolaides's study?
Centola's study uses custom-made data from a controlled online experiment. The dataset was tailor-made, providing strong internal validity — because participants were randomly assigned to different network structures, the study can establish causation between network topology and behavior spread, making their research question easier to answer. Centola also had complete data and full control over variables, eliminating algorithmic confounding. On the downside, the sample was small (~1,500 people), the network was artificial, and participants knew they were in an experiment, introducing reactivity (Salganik, 2.3.3) and limiting external validity.*

In Nicolaides' study, they collected a ready-made dataset from a fitness app, saving them time and resources. This gives massive scale, always-on longitudinal data, and nonreactive measurement of real behavior at low cost. The trade-offs are significant though: without random assignment, only correlations can be shown. In order to rule out homophily, for instance, the study had to make use of instrumental variable design to control for confounding variables. The data is also incomplete (no info on motivation or health), potentially algorithmically confounded by app notifications, and nonrepresentative since app users skew wealthier and younger.

### 2. How do you think these differences can influence the interpretation of the results in each study?
The differences in data collection affect how we interpret the results and what we can actually conclude. In Centola’s custom-made experiment, the controlled design and random assignment allow for stronger causal claims, since alternative explanations (e.g., homophily or confounding factors) are minimized by design. Centola can confidently say that network structure caused differences in behavior spread, since he controlled for everything else. But because the setting was artificial and participants knew about the experiment, it is hard to know whether these dynamics would play out in real social networks. The findings are internally solid but may not generalize.*

In contrast, Nicolaides’ ready-made data come from a natural setting, making the results more reflective of real behavior. Nicolaides shows patterns of exercise contagion among real people at large scale, which feels more applicable to the real world. However, causal inference is weaker because the data was not generated through random assignment. This makes him unable to rule out many possible confounders — friends might run more at the same time due to shared weather or app-driven challenges, not genuine social influence. Even with techniques like instrumental variables, unobserved confounding might always remain a small possibility due to missing variables. Therefore, Centola’s results are more convincing in terms of causality, while Nicolaides’ findings may be stronger in terms of real-world relevance but more cautious in causal interpretation.

## Part 2: Find Researchers using the OpenAlex API

In [1]:
import pandas as pd
import requests
import time
import unicodedata
import re

In [3]:
# read the file
names_df = pd.read_csv("cleaned_names.csv")

# read name column
canonical_names = names_df["name"].dropna().astype(str).tolist()

print("Loaded names:", len(canonical_names))
print(canonical_names[:10])

Loaded names: 1550
['Aakriti Kumar', 'Aaron Clauset', 'Aaron D Nichols', 'Aaron Reeves', 'Aaron Schein', 'Abdulaziz Alhumaidy', 'Abdullah Almaatouq', 'Abhisek Dash', 'Abraham Israeli', 'Achim Edelmann']


In [23]:
# test with a random name
test_name = "Duncan Watts"
url = "https://api.openalex.org/authors"
params = {"search": test_name}
response = requests.get(url, params=params)
data = response.json()

print(f"Status: {response.status_code}")
print(f"Number of results: {len(data['results'])}")

Status: 200
Number of results: 25


In [24]:
first = data['results'][0]
print(f"Keys: {first.keys()}\n")
print(f"ID:            {first['id']}")
print(f"Name:          {first['display_name']}")
print(f"Works API URL: {first['works_api_url']}")
print(f"Works count:   {first['works_count']}")
print(f"H-index:       {first['summary_stats']['h_index']}")
print(f"Institutions:  {first.get('last_known_institutions', [])}")

Keys: dict_keys(['id', 'orcid', 'display_name', 'display_name_alternatives', 'relevance_score', 'works_count', 'cited_by_count', 'summary_stats', 'ids', 'affiliations', 'last_known_institutions', 'topics', 'topic_share', 'x_concepts', 'counts_by_year', 'longest_name', 'parsed_longest_name', 'block_key', 'works_api_url', 'updated_date', 'created_date'])

ID:            https://openalex.org/A5000679279
Name:          Duncan J. Watts
Works API URL: https://api.openalex.org/works?filter=author.id:A5000679279
Works count:   233
H-index:       70
Institutions:  [{'id': 'https://openalex.org/I79576946', 'ror': 'https://ror.org/00b30xv10', 'display_name': 'University of Pennsylvania', 'country_code': 'US', 'type': 'education', 'lineage': ['https://openalex.org/I79576946']}, {'id': 'https://openalex.org/I922845939', 'ror': 'https://ror.org/03zzmyz63', 'display_name': 'Philadelphia University', 'country_code': 'US', 'type': 'education', 'lineage': ['https://openalex.org/I922845939']}]


In [25]:
# why 25 results?
for i, result in enumerate(data['results']):
    print(f"{i+1}. {result['display_name']} | "
          f"relevance: {result['relevance_score']} | "
          f"works: {result['works_count']} | "
          f"h-index: {result['summary_stats']['h_index']}")

1. Duncan J. Watts | relevance: 46806.992 | works: 233 | h-index: 70
2. Duncan J. Watts | relevance: 8282.401 | works: 143 | h-index: 21
3. Andrew Geddes | relevance: 6774.573 | works: 218 | h-index: 35
4. Juliet Lodge | relevance: 4101.6895 | works: 219 | h-index: 20
5. Simon Usherwood | relevance: 2905.7786 | works: 153 | h-index: 15
6. Duncan Watts | relevance: 1521.3883 | works: 29 | h-index: 4
7. Duncan J. Watts | relevance: 1203.4829 | works: 1 | h-index: 1
8. Joseph W. Duncan | relevance: 637.69574 | works: 35 | h-index: 5
9. Duncan J. Watts | relevance: 559.132 | works: 2 | h-index: 2
10. Duncan Watts | relevance: 396.17896 | works: 36 | h-index: 1
11. DUNCAN J. WATTS | relevance: 332.79126 | works: 1 | h-index: 1
12. Duncan Watts | relevance: 293.11493 | works: 2 | h-index: 1
13. Duncan Watts | relevance: 292.1522 | works: 2 | h-index: 1
14. Duncan Watts | relevance: 292.1522 | works: 2 | h-index: 1
15. Duncan J. Watts | relevance: 166.60385 | works: 3 | h-index: 1
16. Duncan 

### Obervations from test request
The API returns 25 results for a single name. Either OpenAlex stores duplicates or the same individual or they are distinct individuals. Some results dont match the searched name, likely due to matching on collaborators or metadata.
However, the relevance score drops sharply after the top result, which suggest the first result is probably what we should go with.

Compare similar names:

In [4]:
def norm(s: str) -> str:
    s = unicodedata.normalize("NFKD", str(s))
    s = "".join(c for c in s if not unicodedata.combining(c))
    s = s.replace(".", "").replace("’", "'").replace("´", "'")
    s = re.sub(r"\s+", " ", s).strip().lower()
    return s

def candidate_best_name_match_score(query_name: str, cand: dict) -> int:
    q = norm(query_name)

    names_to_compare = [cand.get("display_name", "")]
    names_to_compare += cand.get("display_name_alternatives") or []

    # exact match on any representation wins
    for n in names_to_compare:
        if norm(n) == q:
            return 100

    # score = how many tokens overlap
    q_tokens = set(q.split())
    best_overlap = 0
    for n in names_to_compare:
        t = set(norm(n).split())
        best_overlap = max(best_overlap, len(q_tokens & t))

    return best_overlap


Fetch info

In [7]:
session = requests.Session()
session.headers.update({"User-Agent": "Mozilla/5.0 (IC2S2 name lookup)"})

PER_PAGE = 5
OPENALEX_API_KEY = "NkafC7tV5c7anrujEZT7xI".strip()

def fetch_author_row(query_name: str) -> dict:
    url = "https://api.openalex.org/authors"
    params = {
        "search": query_name,
        "per-page": PER_PAGE,
        "select": ",".join([
            "id",
            "display_name",
            "display_name_alternatives",
            "works_api_url",
            "works_count",
            "summary_stats",
            "last_known_institutions"
        ]),
        "api_key": OPENALEX_API_KEY
    }

    r = session.get(url, params=params, timeout=30)
    r.raise_for_status()
    data = r.json()
    results = data.get("results", [])

    if not results:
        return {
            "query_name": query_name,
            "id": None,
            "display_name": None,
            "works_api_url": None,
            "h_index": None,
            "works_count": None,
            "country_code": None,
            "matched": False
        }

    best = max(results, key=lambda c: candidate_best_name_match_score(query_name, c))

    # h-index lives in summary_stats
    summary = best.get("summary_stats") or {}
    h_index = summary.get("h_index")

    # last_known_institutions is a LIST; take the first one (most recent)
    insts = best.get("last_known_institutions") or []
    country_code = insts[0].get("country_code") if insts else None

    return {
        "query_name": query_name,
        "id": best.get("id"),
        "display_name": best.get("display_name"),
        "works_api_url": best.get("works_api_url"),
        "h_index": h_index,
        "works_count": best.get("works_count"),
        "country_code": country_code,
        "matched": True
    }

Run code:

In [ ]:
SLEEP = 0.12
rows = []

for i, name in enumerate(canonical_names, 1):
    try:
        rows.append(fetch_author_row(name))
    except Exception as e:
        rows.append({
            "query_name": name,
            "id": None,
            "display_name": None,
            "works_api_url": None,
            "h_index": None,
            "works_count": None,
            "country_code": None,
            "matched": False,
            "error": str(e)
        })
    # Print progress
    if i % 25 == 0:
        print(f"Processed {i}/{len(canonical_names)}")
    time.sleep(SLEEP)

authors_df = pd.DataFrame(rows)
authors_df.to_csv("openalex_authors_from_cleaned_names.csv", index=False) # Save list
authors_df.head()

Processed 25/1550
Processed 50/1550
Processed 75/1550
Processed 100/1550
Processed 125/1550
Processed 150/1550
Processed 175/1550
Processed 200/1550
Processed 225/1550
Processed 250/1550
Processed 275/1550
Processed 300/1550
Processed 325/1550
Processed 350/1550
Processed 375/1550
Processed 400/1550
Processed 425/1550
Processed 450/1550
Processed 475/1550
Processed 500/1550
Processed 525/1550
Processed 550/1550
Processed 575/1550
Processed 600/1550
Processed 625/1550
Processed 650/1550
Processed 675/1550
Processed 700/1550
Processed 725/1550
Processed 750/1550
Processed 775/1550
Processed 800/1550
Processed 825/1550
Processed 850/1550
Processed 875/1550
Processed 900/1550
Processed 925/1550
Processed 950/1550
Processed 975/1550
Processed 1000/1550
Processed 1025/1550
Processed 1050/1550
Processed 1075/1550
Processed 1100/1550
Processed 1125/1550
Processed 1150/1550
Processed 1175/1550
Processed 1200/1550
Processed 1225/1550
Processed 1250/1550
Processed 1275/1550
Processed 1300/1550
Pr

,query_name,id,display_name,works_api_url,h_index,works_count,country_code,matched,error
0,Aakriti Kumar,https://openalex.org/A5040234818,Aakriti Kumar,https://api.openalex.org/works?filter=author.i...,9.0,20.0,US,True,NaN
1,Aaron Clauset,https://openalex.org/A5014647140,Aaron Clauset,https://api.openalex.org/works?filter=author.i...,47.0,267.0,US,True,NaN
2,Aaron D Nichols,https://openalex.org/A5089395967,Aaron Nichols,https://api.openalex.org/works?filter=author.i...,3.0,10.0,US,True,NaN
3,Aaron Reeves,https://openalex.org/A5039995516,Aaron Reeves,https://api.openalex.org/works?filter=author.i...,48.0,254.0,GB,True,NaN
4,Aaron Schein,https://openalex.org/A5021334672,Joshua Maynez,https://api.openalex.org/works?filter=author.i...,12.0,51.0,NaN,True,NaN


### Reflection Questions

- **Which challenges did you encounter? How did you address them?**

To match ic2s2 names to openalex authors, we used the search endpoint and retrieved the top 5 candidates per name. since the api often returns multiple people for the same query, we implemented a custom scoring function that normalizes names (removing accents, dots, extra whitespace) and scores candidates by token overlap between the queried name and the candidate's display name or alternatives. the best-scoring candidate is selected as the match.

- **Choose one problem you faced while collecting the data and describe your solution. Why did you choose this approach, and what impact might it have on your data?**

This approach handles common variations like "Aaron D Nichols" vs "Aaron Nichols" well, but has limitations. token overlap can't distinguish between people who share the same name, and it may match the wrong person entirely if no good candidate exists. a relevance score cutoff could help filter out poor matches. we also hit 429 rate limit errors during fetching, which we resolved by adding a 0.12 second delay between requests.


## Part 3: Collect Research Articles

In [5]:
import pandas as pd

ic2s2_authors = pd.read_csv("openalex_authors_from_cleaned_names.csv")
ic2s2_authors = ic2s2_authors.dropna(subset=["id"])
print(f"total authors: {len(ic2s2_authors)}")
print(ic2s2_authors['works_count'].describe())

# filter authors: works_count between 5 and 5000
ic2s2_authors = ic2s2_authors[(ic2s2_authors['works_count'] >= 5) & (ic2s2_authors['works_count'] <= 5000)]
print(f"authors after filtering: {len(ic2s2_authors)}")

total authors: 1475
count    1475.000000
mean      114.425763
std       216.095205
min         1.000000
25%        12.000000
50%        46.000000
75%       126.500000
max      2790.000000
Name: works_count, dtype: float64
authors after filtering: 1287


### Exploring the API

In [6]:
import requests

api_key = "zhj01t4KDUqWAHNFrcZrok"

# pick one author to test with
sample_id = ic2s2_authors['id'].iloc[1]
print(f"testing with: {sample_id}")

response = requests.get(
    "https://api.openalex.org/works",
    params={
        "api_key": api_key,
        "filter": f"author.id:{sample_id}",
        "per_page": 1
    }
)

data = response.json()
print(f"total works for this author: {data['meta']['count']}")
print(f"cost of this call: ${data['meta']['cost_usd']}")

testing with: https://openalex.org/A5014647140
total works for this author: 267
cost of this call: $0.0001


In [7]:
work = data['results'][0]
print(work.keys())

dict_keys(['id', 'doi', 'title', 'display_name', 'publication_year', 'publication_date', 'ids', 'language', 'primary_location', 'type', 'indexed_in', 'open_access', 'authorships', 'institutions', 'countries_distinct_count', 'institutions_distinct_count', 'corresponding_author_ids', 'corresponding_institution_ids', 'apc_list', 'apc_paid', 'fwci', 'has_fulltext', 'cited_by_count', 'citation_normalized_percentile', 'cited_by_percentile_year', 'biblio', 'is_retracted', 'is_paratext', 'is_xpac', 'primary_topic', 'topics', 'keywords', 'concepts', 'mesh', 'locations_count', 'locations', 'best_oa_location', 'sustainable_development_goals', 'awards', 'funders', 'has_content', 'content_urls', 'referenced_works_count', 'referenced_works', 'related_works', 'abstract_inverted_index', 'counts_by_year', 'updated_date', 'created_date'])


In [8]:
print("id:", work['id'])
print("publication_year:", work['publication_year'])
print("cited_by_count:", work['cited_by_count'])
print("title:", work['title'])
print()
print("authorships (first entry):", work['authorships'][0])
print()
print("abstract_inverted_index (first 3 items):", dict(list(work['abstract_inverted_index'].items())[:3]) if work['abstract_inverted_index'] else "None")

id: https://openalex.org/W2047940964
publication_year: 2004
cited_by_count: 7334
title: Finding community structure in very large networks

authorships (first entry): {'author_position': 'first', 'author': {'id': 'https://openalex.org/A5014647140', 'display_name': 'Aaron Clauset', 'orcid': 'https://orcid.org/0000-0002-3529-8746'}, 'institutions': [{'id': 'https://openalex.org/I169521973', 'display_name': 'University of New Mexico', 'ror': 'https://ror.org/05fs6jp91', 'country_code': 'US', 'type': 'education', 'lineage': ['https://openalex.org/I169521973']}], 'countries': ['US'], 'is_corresponding': True, 'raw_author_name': 'Aaron Clauset', 'raw_affiliation_strings': ['Department of Computer Science, University of New Mexico, Albuquerque, NM 87131, USA', 'Department of Computer Science, University of New Mexico, Albuquerque, New Mexico 87131, USA'], 'affiliations': [{'raw_affiliation_string': 'Department of Computer Science, University of New Mexico, Albuquerque, NM 87131, USA', 'instit

In [9]:
author_ids = [a['author']['id'] for a in work['authorships']]
print(f"number of authors: {len(author_ids)}")
print(author_ids)

number of authors: 3
['https://openalex.org/A5014647140', 'https://openalex.org/A5067142016', 'https://openalex.org/A5008033989']


### Building the filters

In [10]:
response = requests.get(
    "https://api.openalex.org/works",
    params={
        "api_key": api_key,
        "filter": f"author.id:{sample_id},cited_by_count:>10",
        "per_page": 1
    }
)

data = response.json()
print(f"works with >10 citations: {data['meta']['count']}")

works with >10 citations: 75


In [11]:
concepts_to_find = ["Sociology", "Psychology", "Economics", "Political Science", 
                     "Mathematics", "Physics", "Computer Science"]

for c in concepts_to_find:
    r = requests.get(
        "https://api.openalex.org/concepts",
        params={"api_key": api_key, "filter": f"display_name.search:{c},level:0"}
    )
    results = r.json()['results']
    if results:
        print(f"{c}: {results[0]['id']}")
    else:
        print(f"{c}: NOT FOUND")

Sociology: https://openalex.org/C144024400
Psychology: https://openalex.org/C15744967
Economics: https://openalex.org/C162324750
Political Science: https://openalex.org/C17744445
Mathematics: https://openalex.org/C33923547
Physics: https://openalex.org/C121332964
Computer Science: https://openalex.org/C41008148


In [12]:
social = "C144024400|C15744967|C162324750|C17744445"
quant = "C33923547|C121332964|C41008148"

response = requests.get(
    "https://api.openalex.org/works",
    params={
        "api_key": api_key,
        "filter": f"author.id:{sample_id},cited_by_count:>10,concepts.id:{social},concepts.id:{quant}",
        "per_page": 1
    }
)

data = response.json()
print(f"works matching all filters: {data['meta']['count']}")

works matching all filters: 32


In [13]:
top_authors = ic2s2_authors.nlargest(5, 'h_index')[['display_name', 'id', 'h_index']]
print(top_authors)

       display_name                                id  h_index
695         Li Song  https://openalex.org/A5100448217    142.0
982    Michaël Maes  https://openalex.org/A5032880777    141.0
1178    Robert West  https://openalex.org/A5059645286    132.0
675   Qinghua Zhang  https://openalex.org/A5100327261    129.0
43    Alex Pentland  https://openalex.org/A5007176508    121.0


In [14]:
pentland_id = "https://openalex.org/A5007176508"

response = requests.get(
    "https://api.openalex.org/works",
    params={
        "api_key": api_key,
        "filter": f"author.id:{pentland_id},cited_by_count:>10,concepts.id:{social},concepts.id:{quant}",
        "per_page": 1
    }
)

data = response.json()
print(f"Pentland works matching all filters: {data['meta']['count']}")

Pentland works matching all filters: 252


### Fetching all works

In [15]:
batch = ic2s2_authors['id'].iloc[:25]
author_filter = "|".join(batch.values)

response = requests.get(
    "https://api.openalex.org/works",
    params={
        "api_key": api_key,
        "filter": f"author.id:{author_filter},cited_by_count:>10,concepts.id:{social},concepts.id:{quant}",
        "per_page": 1
    }
)

data = response.json()
print(f"works from first 25 authors: {data['meta']['count']}")
print(f"cost of this call: ${data['meta']['cost_usd']}")

works from first 25 authors: 171
cost of this call: $0.0001


In [16]:
import math

n_batches = math.ceil(len(ic2s2_authors) / 25)
est_total_works = (len(ic2s2_authors) / 25) * data['meta']['count']
est_pages = est_total_works / 100
est_calls = n_batches + est_pages
est_cost = est_calls * 0.0001

print(f"number of batches: {n_batches}")
print(f"estimated total works: {est_total_works:.0f}")
print(f"estimated total API calls: {est_calls:.0f}")
print(f"estimated cost: ${est_cost:.2f}")

number of batches: 52
estimated total works: 8803
estimated total API calls: 140
estimated cost: $0.01


In [17]:
def fetch_works_for_batch(author_ids, api_key, social, quant):
    """fetch all works for a batch of authors, handling pagination with cursor."""
    all_works = []
    author_filter = "|".join(author_ids)
    cursor = "*"
    
    while cursor:
        response = requests.get(
            "https://api.openalex.org/works",
            params={
                "api_key": api_key,
                "filter": f"author.id:{author_filter},cited_by_count:>10,concepts.id:{social},concepts.id:{quant}",
                "per_page": 100,
                "cursor": cursor
            }
        )
        data = response.json()
        all_works.extend(data.get('results', []))
        cursor = data['meta'].get('next_cursor')
    
    return all_works

# test on first batch
batch = ic2s2_authors['id'].iloc[:25].values
works = fetch_works_for_batch(batch, api_key, social, quant)
print(f"fetched {len(works)} works from first batch")

fetched 171 works from first batch


In [19]:
from tqdm import tqdm
from joblib import Parallel, delayed

all_ids = ic2s2_authors['id'].values
batches = [all_ids[i:i+25] for i in range(0, len(all_ids), 25)]
print(f"total batches: {len(batches)}")

all_works_nested = Parallel(n_jobs=5)(
    delayed(fetch_works_for_batch)(batch, api_key, social, quant)
    for batch in tqdm(batches)
)

all_works = [w for batch_works in all_works_nested if batch_works for w in batch_works]
print(f"total works fetched: {len(all_works)}")

unique_ids = set(w['id'] for w in all_works)
print(f"unique works: {len(unique_ids)}")

total batches: 52


100%|██████████| 52/52 [01:01<00:00,  1.18s/it]


total works fetched: 16837
unique works: 14954


### Processing and saving

In [20]:
papers = []
abstracts = []

for w in all_works:
    author_ids = [a['author']['id'] for a in w['authorships']]
    
    if len(author_ids) >= 10:
        continue
    
    papers.append({
        'id': w['id'],
        'publication_year': w['publication_year'],
        'cited_by_count': w['cited_by_count'],
        'author_ids': author_ids
    })
    
    abstracts.append({
        'id': w['id'],
        'title': w['title'],
        'abstract_inverted_index': w['abstract_inverted_index']
    })

print(f"works after filtering <10 authors: {len(papers)} (removed {len(all_works) - len(papers)})")

df_papers = pd.DataFrame(papers).drop_duplicates(subset='id')
df_abstracts = pd.DataFrame(abstracts).drop_duplicates(subset='id')

print(f"D2 - IC2S2 papers: {len(df_papers)}")
print(f"D3 - IC2S2 abstracts: {len(df_abstracts)}")

works after filtering <10 authors: 15529 (removed 1308)
D2 - IC2S2 papers: 13950
D3 - IC2S2 abstracts: 13950


In [21]:
df_papers.to_json("ic2s2_papers.json", orient='records')
df_abstracts.to_json("ic2s2_abstracts.json", orient='records')

print("D2 saved to ic2s2_papers.json")
print("D3 saved to ic2s2_abstracts.json")

D2 saved to ic2s2_papers.json
D3 saved to ic2s2_abstracts.json


In [22]:
all_author_ids = set()
for ids in df_papers['author_ids']:
    all_author_ids.update(ids)

print(f"unique works: {len(df_papers)}")
print(f"unique researchers who co-authored these works: {len(all_author_ids)}")

unique works: 13950
unique researchers who co-authored these works: 21796


### Data Overview and Reflection Questions

- **Dataset summary:**

The ic2s2 papers dataset (D2) contains 13,950 unique works. across these works, there are 21,796 unique researchers who have co-authored at least one paper.

- **Efficiency in code:**

We used several strategies to speed up data collection. first, we batched 25 author IDs per API request using the OR filter syntax, reducing ~1,287 individual calls to just 52. second, we applied filters directly in the API call (cited_by_count >10, concept filters for social science + quantitative fields) so only relevant works were returned. third, we set per_page=100 to minimize pagination calls. finally, we used joblib's Parallel with 5 cores to run batches simultaneously. combined, these brought total fetch time down to about 1 minute.

- **Filtering criteria and dataset relevance:**

The filters help keep the dataset focused on established, interdisciplinary css research, but they introduce biases. the >10 citations threshold excludes recent work that hasn't had time to accumulate citations, underrepresenting emerging topics and early-career researchers. the concept filter requiring both a social science and a quantitative field excludes qualitative css work like computational ethnography. the <10 authors filter removes large collaborative projects, which are increasingly common. the works_count 5–5000 filter removes very inactive researchers but also potentially excludes newcomers. overall, the dataset skews toward well-established, quantitative, small-team research.